language/vm/vm_runtime/src/execution_stack.rs

```diff
@@ -40,7 +41,11 @@ where
     pub fn push_call(&mut self, function: FunctionRef<'txn>) -> VMResult<()> {
         let callee_arg_size = function.arg_count();
         let args = self.popn(callee_arg_size as u16)?;
-        self.function_stack.push(Frame::new(function, args));
+        if self.function_stack.len() < usize::max_value() {
```

Comment:
We are most likely going to run out memory long before the length overflows. We could appeal to gas limits to justify an assumption, or impose a more sensible limit on the function stack size.